feature_engine/timeseries/forecasting/base_forecast_transformers.py

```diff
@@ -78,9 +73,26 @@ def __init__(
                 f"Got {drop_original} instead."
             )
 
+        # check validity if group by variables passed
+        if group_by_variables:
+            # check group by variables data-types
+            if not (
+                isinstance(group_by_variables, str)
+                or isinstance(group_by_variables, list)
+            ):
+                raise ValueError(
+                    "group_by_variables must be an string or a list of strings. "
+                    f"Got {group_by_variables} instead."
+                )
+            # check if passed list has duplicates.
+            if isinstance(group_by_variables, list):
+                if len(set(group_by_variables)) != len(group_by_variables):
+                    raise ValueError("group_by_variables contains duplicate values")
+
```

Comment:
To improve the code I will do the following: after the checks on `group_by_variables`, I will put this variable into a list if the variable is just a string. In this way from now on you know that you are working with a list of string and you don't need to do other checks (like for example the one at line 180).

You could do something like

```python
if group_by_variables:
             if isinstance(group_by_variables, str):
                      self.group_by_variables = [group_by_variables]
             elif not (
                      isinstance(group_by_variables, list) and
                      all(isinstance(element, str) for element in group_by_variables)
             ):
                       raise ValueError(
                              "group_by_variables must be an string or a list of strings. "
                              f"Got {group_by_variables} instead."
                         )
            else:
                     # note that if you are here, then group_by_variables is a list
                     if len(set(group_by_variables)) != len(group_by_variables):
                     raise ValueError("group_by_variables contains duplicate 
```

Reply:
Unfortunately, for consistancy with the sklearn api we cannot modify the input parameters. We need to leave them as they are.